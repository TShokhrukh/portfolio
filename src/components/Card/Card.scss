.Card {
  --card-border-color: #{rgba($purple, 0.5)};
  --card-border-radius: #{$border-radius * 2};
  --card-padding-x: #{$border-radius * 2.5};
  --card-padding-y: #{$border-radius * 3.75};
  border-radius: var(--card-border-radius);
  border: 1px solid var(--card-border-color);
  padding: var(--card-padding-y) var(--card-padding-x);
  transition: 0.2s;

  &:hover {
    --card-border-color: transparent;
    background-color: rgba($purple, 0.1);
  }
}