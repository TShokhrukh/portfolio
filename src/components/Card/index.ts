export * from './build'
export * from './__types__'
export { Card as CardBase } from './Card'
