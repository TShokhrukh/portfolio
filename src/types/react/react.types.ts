import type { ElementType } from 'react'

export interface IComponentProp<P = any> {
  component?: ElementType<P>,
}
