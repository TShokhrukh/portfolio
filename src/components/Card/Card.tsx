import React, { FC } from 'react'
import { cnCard } from './Card.const'
import type { ICardProps } from './__types__'
import './Card.scss'

export const Card: FC<ICardProps> = ({
  component: Component = 'div',
  className,
  children,
  ...props
}) => {
  return (
    <Component {...props} className={cnCard({}, [className])}
      >{children}</Component>
  )
}
