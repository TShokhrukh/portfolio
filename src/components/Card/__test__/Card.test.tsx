/* global test, jest, expect, describe */
import React from 'react'
import { fireEvent, render, screen } from '@testing-library/react'
import { Card } from '../Card'

describe('Card', () => {
  test('must return the component template', () => {
    render(<Card data-testid='card'>children</Card>)

    expect(screen.getByTestId('card')).toMatchSnapshot()
  })

  test('must change tag name by component prop', () => {
    render(<Card component='article' data-testid='card' />)

    expect(screen.getByTestId('card').tagName).toBe('ARTICLE')
  })
})
