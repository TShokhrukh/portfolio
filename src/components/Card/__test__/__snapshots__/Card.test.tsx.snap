// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card must return the component template 1`] = `
<div
  class="Card"
  data-testid="card"
>
  children
</div>
`;
